### PR TITLE
feat(redis-interface): add redis interface command to set multiple the keys in redis and increment if the key already exists

### PR DESCRIPTION
**Pull request by Aprabhat19**
## Type of Change

- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description
add redis interface command to set multiple the keys in redis and increment if the key already exists

### Additional Changes

- [ ] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## How did you test it?
 
 - Wrote a unit test to insert the keys
Screenshot 2024-12-13 at 1 05 33 PM

## Checklist

- [ ] I formatted the code `cargo +nightly fmt --all`
- [ ] I addressed lints thrown by `cargo clippy`
- [ ] I reviewed the submitted code
- [ ] I added unit tests for my changes where possible

### CODE REVIEW

File: crates/redis_interface/src/commands.rs [hunk 1]
```diff
@@ -852,12 +852,28 @@ impl super::RedisConnectionPool {
             .await
             .change_context(errors::RedisError::ConsumerGroupClaimFailed)
     }
+
+    #[instrument(level = "DEBUG", skip(self))]
+    pub async fn set_keys_using_script(
```
Comment (NishantJoshi00) [inline crates/redis_interface/src/commands.rs:857]: this is not a set interface, rather an `incr` interface.

File: crates/redis_interface/src/commands.rs [hunk 2]
```diff
@@ -852,12 +852,28 @@ impl super::RedisConnectionPool {
...
+        &self,
+        keys: Vec<String>,
+        values: Vec<String>,
```
Comment (NishantJoshi00) [inline crates/redis_interface/src/commands.rs:860]: Have we evaluated the case if `keys` and `values` have different length?
in that case, better write this as `key_values: Vec<(String, usize)>`

**Review by NishantJoshi00 — CHANGES_REQUESTED** [review]

File: crates/redis_interface/src/commands.rs (hunk 1 above)
Comment (Aprabhat19) [reply crates/redis_interface/src/commands.rs:857]: Sure will change the name

**Review by NishantJoshi00 — APPROVED** [review]

File: crates/redis_interface/src/commands.rs [hunk 3]
```diff
@@ -852,12 +852,32 @@ impl super::RedisConnectionPool {
             .await
             .change_context(errors::RedisError::ConsumerGroupClaimFailed)
     }
+
+    #[instrument(level = "DEBUG", skip(self))]
+    pub async fn incr_keys_using_script(
+        &self,
+        lua_script: &str,
+        keys_and_values: HashMap<String, usize>,
```
Comment (NishantJoshi00) [inline crates/redis_interface/src/commands.rs:860]: Separate this if the script is generic

**Review by NishantJoshi00 — CHANGES_REQUESTED** [review]

File: crates/redis_interface/src/commands.rs [hunk 4]
```diff
@@ -852,12 +852,31 @@ impl super::RedisConnectionPool {
             .await
             .change_context(errors::RedisError::ConsumerGroupClaimFailed)
     }
+
+    #[instrument(level = "DEBUG", skip(self))]
+    pub async fn incr_keys_using_script<V>(
+        &self,
+        lua_script: &str,
+        key: Vec<String>,
```
Comment (NishantJoshi00) [inline crates/redis_interface/src/commands.rs:860]: Can't this be multiple keys?

**Review by NishantJoshi00 — DISMISSED** [review]

File: crates/redis_interface/src/commands.rs (hunk 4 above)
Comment (Aprabhat19) [reply crates/redis_interface/src/commands.rs:860]: Can be but do we also want to make the keys generic,I mean should we allow that? As today in our redis-interface module there's no other key than a string

**Review by Sarthak1799 — DISMISSED** [review]

**Review by NishantJoshi00 — APPROVED** [review]

**Review by Sarthak1799 — APPROVED** [review]

**Pull request merged**